apidump/net.go

```diff
@@ -54,12 +55,36 @@ func showPermissionErrors(sampleError error) error {
 			printer.Warningf("The agent needs the CAP_NET_RAW capability to capture packets. You are running as an unprivileged (non-root) user.\n")
 			return errors.Errorf("Insufficient permissions, try using \"sudo\" to run as root.")
 		}
+	} else if strings.Contains(sampleError.Error(), "SIOCETHTOOL(ETHTOOL_GET_TS_INFO) ioctl failed: Function not implemented") {
+		// This happens when the binary was built for a different architecture, e.g.
+		// if the user pulled the amd64 Docker image on arm64.
+		arch := architecture.GetCanonicalArch()
+
+		printer.Warningf(
+			"The agent received \"Function not implemented\" when trying to read from your network interfaces. "+
+				"This often indicates that the Akita agent was built for a different architecture than your host architecture. "+
+				"This Akita agent binary was built for %s.\n",
+			arch,
+		)
+
+		if os.Getenv("__X_AKITA_CLI_DOCKER") == "true" {
+			return errors.Errorf(
```

Comment:
Do we want to distinguish this in the client telemetry?

https://github.com/akitasoftware/akita-cli/blob/d94be84ef8c80a2c1de3b73b9e704bf2af755471/apidump/apidump.go#L431-L436

Right now it is listed as a permission error but we could add a new cause to our enum (and then handle it on the backend...)  Our Segment events will have the raw errors, so we'll see it either way, but the user can't if looking at the UI.  (I don't recall exactly what they see, do we show the original error message or just one based on type?)

Reply:
This is what we show for that error type, it would be good to define a new type or pick OTHER, which does display the original.
```
        return (
          <Warning
            title="Client Lacks Permissions"
            message={`The Akita client could not capture traffic because it lacks permissions to do so. In a container environment, ensure that the container which has the client can run in privileged mode and has the CAP_NET_RAW capability; this might be disabled by default, or not permitted by your service provider. If running directly on a host or VM, start the client as the root user, or set the CAP_NET_RAW capability on the binary. Please contact support@akitasoftware.com with a description of your runtime environment if you need further assistance.`}
          />
        );
```

Reply:
Good idea.  I'm going to leave this PR open while I add support for a new telemetry error to the backend, and then add that here.